EditPendingList use case (without screens)

SharonHsieh22 opened the pull request:
EDIT: November 18, 2022

- EditPendingList and ViewPendingList use cases are basically finished
- Javadocs and method descriptions added (I formatted them like Aarya's)
- Added addGroup, removeFromApplications to User entity
- Added removeFromRequests, addMember to Group entity

Working on the pending list screen now.

*please ignore the the 11 commits I made, I was having merge conflicts but I finally learned how to solve those... if anyone has the same problem I might be able to help 

**also I accidently commited the files I pulled from main, which I don't think is necessary(?) I'm proabably going to close this pull request and make a new one later

soheegoo committed:
First initial version of the Group Creation Use Case. Just wanted to commit to show you guys what I've done so far. Probably more things to change/add

soheegoo committed:
Added main file which can open up Java Swing UI for my Group Register Screen. However, there's no data persistence yet and clicking the buttons don't do anything. Confused on how to switch screens.

SharonHsieh22 committed:
Added addGroup, removeFromApplications to User entity.
Added removeFromRequests, addMember to Group entity.

SharonHsieh22 committed:
Added addGroup, removeFromApplications to User entity.
Added removeFromRequests, addMember to Group entity.

SharonHsieh22 committed:
Added addGroup, removeFromApplications to User entity.
Added removeFromRequests, addMember to Group entity.

aaryavatsa reviewed (dismissed):
Added entity functions look good!

ipekakyol reviewed (dismissed):
Both the changes and the newly added methods seem necessary and well done.

SharonHsieh22 committed:
Added the main functions of feature 8 (edit pending list use case). TODO: create screens and edit EditPendingListResponseModel to contain the necessary information needed for screen creation.

SharonHsieh22 committed:
Added the main functions of feature 8 (edit pending list use case). TODO: create screens and edit EditPendingListResponseModel to contain the necessary information needed for screen creation.

SharonHsieh22 committed:
Added the main functions of feature 8 (edit pending list use case). TODO: create screens and edit EditPendingListResponseModel to contain the necessary information needed for screen creation.

SharonHsieh22 committed:
Added the main functions of feature 8 (edit pending list use case). TODO: create screens and edit EditPendingListResponseModel to contain the necessary information needed for screen creation.

SharonHsieh22 committed:
Added the main functions of feature 8 (edit pending list use case). TODO: create screens and edit EditPendingListResponseModel to contain the necessary information needed for screen creation.

SharonHsieh22 committed:
Added the main functions of feature 8 (edit pending list use case). TODO: create screens and edit EditPendingListResponseModel to contain the necessary information needed for screen creation.

aaryavatsa commented on src/main/java/edit_pending_list_use_case/EditPendingListInteractor.java:
```diff
@@ -0,0 +1,50 @@
+
+        if(!dsGateway.userExistsByID(userID)) {return presenter.prepareFailView("User does not exist.");}
+
+        // TODO: this shouldn't be possible, maybe make it an exception instead of presenting it to the user?
```
You're so right about this actually.. in my use case I also checked for these weird border cases even though they shouldn't be possible. Would it be better to make them exceptions? Having these explicit checks makes my interactor code a lot longer.

soheegoo replied:
Yes, I think having them as exceptions is a good idea.

aaryavatsa reviewed (dismissed):
Added functions in the entity classes look good. Use case looks good too! Easy to follow along and very in-line with the clean architecture principles we've learned in class.

SharonHsieh22 committed:
Added the main functions of feature 8 (edit pending list use case). TODO: create screens and edit EditPendingListResponseModel to contain the necessary information needed for screen creation.

SharonHsieh22 committed:
Added the main functions of feature 8 (edit pending list use case). TODO: create screens and edit EditPendingListResponseModel to contain the necessary information needed for screen creation.

SharonHsieh22 committed:
Added the main functions of feature 8 (edit pending list use case). TODO: create screens and edit EditPendingListResponseModel to contain the necessary information needed for screen creation.

ileuxali committed:
Made 2 changes to User class:
1. made it an interface to allow multiple types of users, currently just normal user
2. deleted userID attribute, that attribute is giving us a lot of difficulty implementing, instead now we assume username attribute is unique

Also finished the implementation of the singleton class CurrentUser following Sohee's started code to store the current logged in user. Also included a test file to showcase how to use the singleton class.

ileuxali committed:
I messed up earlier, the hashmap for groups should be from String to Group not Integer to Group :^)

aaryavatsa committed:
Merge branch 'main' into feature-1-user-signuplogin

ileuxali committed:
Update Group.java

ileuxali committed:
Create ObjectMap.java

ileuxali committed:
Merge pull request #20 from CSC207-2022F-UofT/feature-1-user-signuplogin

Made changes to user and completed singleton class

SharonHsieh22 committed:
Added initial code for the pending list UI. Changed my use case to reflect the changes made to the user and group entities (got rid of userID and groupID). Added GetPendingList classes to create the pending list UI.

SharonHsieh22 committed:
Added initial code for the pending list UI. Changed my use case to reflect the changes made to the user and group entities (got rid of userID and groupID). Added GetPendingList classes to create the pending list UI.

SharonHsieh22 committed:
Added initial code for the pending list UI. Changed my use case to reflect the changes made to the user and group entities (got rid of userID and groupID). Added GetPendingList classes to create the pending list UI.

soheegoo committed:
Added main file which can open up Java Swing UI for my Group Register Screen. However, there's no data persistence yet and clicking the buttons don't do anything. Confused on how to switch screens.

soheegoo committed:
Added main file which can open up Java Swing UI for my Group Register Screen. However, there's no data persistence yet and clicking the buttons don't do anything. Confused on how to switch screens.

soheegoo committed:
Group Creation Use Case. Including serialization and test files for serialization. Didn't include GroupProfile into the Group Entity because I need Julia's GroupProfile Entity first, or else it give me an error since the class doesn't exist yet.

ileuxali committed:
deleted a useless test

Deleted a useless test. It was made during initial testing stages to show progress. It should've been deleted long ago.

ileuxali committed:
Merge pull request #22 from CSC207-2022F-UofT/1-feature-4-group-creation

1 feature 4 group creation

SharonHsieh22 committed:
Getting changes from main.

SharonHsieh22 committed:
Renamed presenter interface, finished implementing the ViewPendingList use case, added Javadocs and method comments.

SharonHsieh22 committed:
Renamed presenter interface, finished implementing the ViewPendingList use case, added Javadocs and method comments.
